src/main/java/teammates/common/util/ActivityLogGenerator.java

```diff
@@ -0,0 +1,336 @@
+package teammates.common.util;
+
+import java.text.SimpleDateFormat;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.Map;
+import java.util.TimeZone;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import com.google.appengine.api.log.AppLogLine;
+
+import teammates.common.datatransfer.AccountAttributes;
+import teammates.common.datatransfer.StudentAttributes;
+import teammates.common.datatransfer.UserType;
+import teammates.common.exception.TeammatesException;
+
+/**
+ * Factory to generate ActivityLogEntry
+ * 
+ * @see {@link ActivityLogEntry}
+ */
+public class ActivityLogGenerator {
```

Comment:
I put methods together if there are related (e.g. one method is used in another method). If this is confusing, I can change. :)